libqtile/widget/battery.py

```diff
@@ -353,6 +355,11 @@ def poll(self) -> str:
         except RuntimeError as e:
             return 'Error: {}'.format(e)
 
+        if self.notify_below:
+            percent = int(status.percent * 100)
+            if percent < self.notify_below:
+                send_notification("Warning", "Battery at {0}%".format(percent), urgent=True)
```

Comment:
So it's going to send a notification every time the widget is polled? Shouldn't we keep a state to know if it has been sent already?

Reply:
That was the idea but your way would also work. My logic was that if you have low battery you probably don't want to forget! But yes, we could remember if we've sent it, then reset that state when we go back above the threshold.

Reply:
Code looks fine to me, feel free to merge when you guys agree on this :)

Reply:
It's updated so it will only notify the first time.